### PR TITLE
Segmented button

### PR DESCRIPTION
**Pull request by lukeac123**

### CODE REVIEW

File: .changeset/sharp-flowers-relax.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+"@salt-ds/lab": minor
+---
+
+Add `SegmentedButton` to labs
+
+```tsx
+const SegmentedButton = () => (
```
Comment (joshwooding) [inline .changeset/sharp-flowers-relax.md:8]: "SegmentedButton" isn't a good name for what this component is doing. Personally I would rename it to SegmentedButtonGroup or Segmented Group. You should chat with @yunjungyeh

File: .changeset/sharp-flowers-relax.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  <SegmentedButton>
+    <Button variant={variant}> Button </Button>
```
Comment (joshwooding) [inline .changeset/sharp-flowers-relax.md:10]: I would just remove the `variant={variant}` for simplicity.

File: .changeset/sharp-flowers-relax.md (hunk 1 above)
Comment (lukeac123) [reply .changeset/sharp-flowers-relax.md:8]: To me A SegmentedButton suggests a button that is divided into multiple sections so i think it makes sense. Does a SegmentedButtonGroup not suggest a group of buttons all of which are all divided.
@yunjungyeh What do you think ?

File: packages/lab/src/segmented-button/SegmentedButton.css [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+.saltSegmentedButton-separator > *:not(:last-child)::after {
+  content: "";
+  width: 1px;
+  position: absolute;
+  right: 0;
+  padding-top: var(--salt-spacing-100);
+  padding-bottom: var(--salt-spacing-100);
+  background-color: var(--salt-palette-neutral-primary-separator);
```
Comment (joshwooding) [inline packages/lab/src/segmented-button/SegmentedButton.css:13]:
```suggestion
  background: var(--salt-palette-neutral-primary-separator);
```

File: packages/lab/src/segmented-button/SegmentedButton.css [hunk 4]
```diff
@@ -0,0 +1,15 @@
+.saltSegmentedButton {
+  display: flex;
+  flex-direction: row;
+}
+
+.saltSegmentedButton-separator > *:not(:last-child)::after {
+  content: "";
+  width: 1px;
```
Comment (joshwooding) [inline packages/lab/src/segmented-button/SegmentedButton.css:8]: This should be using a size token

File: packages/lab/stories/segmented-button/segmented-button.qa.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import { Button } from "@salt-ds/core";
+import { SegmentedButton, SegmentedButtonProps } from "@salt-ds/lab";
+import { StoryFn, Meta } from "@storybook/react";
+import { QAContainer, QAContainerProps } from "docs/components";
+
+export default {
+  title: "Lab/SegmentedButton/QA",
```
Comment (joshwooding) [inline packages/lab/stories/segmented-button/segmented-button.qa.stories.tsx:7]:
```suggestion
  title: "Lab/Segmented Button/Segmented Button QA", 
```

File: packages/lab/stories/segmented-button/segmented-button.qa.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  return (
+    <SegmentedButton>
+      <Button variant={variant}> Button </Button>
+      <Button variant={variant}> Button </Button>
+      <Button variant={variant}> Button </Button>
+    </SegmentedButton>
+  );
+};
+
+export const Default: StoryFn<QAContainerProps> = () => {
```
Comment (joshwooding) [inline packages/lab/stories/segmented-button/segmented-button.qa.stories.tsx:25]: Is there a benefit having these as separate stories?

File: packages/lab/stories/segmented-button/segmented-button.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
+import { SegmentedButton, SegmentedButtonProps } from "@salt-ds/lab";
+import { ChevronDownIcon, FavoriteIcon } from "@salt-ds/icons";
+import { Button } from "@salt-ds/core";
+import { Meta, StoryFn } from "@storybook/react";
+
+export default {
+  title: "Lab/SegmentedButton",
```
Comment (joshwooding) [inline packages/lab/stories/segmented-button/segmented-button.stories.tsx:7]:
```suggestion
  title: "Lab/Segmented Button",
```

File: packages/lab/stories/segmented-button/segmented-button.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+Secondary.args = {
+  variant: "secondary",
+};
+
+export const CTA = SegmentedButtonTemplate.bind({});
+CTA.args = {
+  variant: "cta",
+};
+
+export const SplitButton: StoryFn<typeof SegmentedButton> = () => {
```
Comment (joshwooding) [inline packages/lab/stories/segmented-button/segmented-button.stories.tsx:38]: I think we should remove this example

File: .changeset/sharp-flowers-relax.md (hunk 1 above)
Comment (joshwooding) [reply .changeset/sharp-flowers-relax.md:8]: I would rather it be called a group. Even If we simplify it and call it a ButtonGroup

**Comment by origami-z** [comment]
Is the current implementation on spec? Hovered button will change background of the separator?

File: packages/lab/src/segmented-button-group/SegmentedButtonGroup.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+  return (
+    <div
+      className={clsx(withBaseName(), withBaseName(`${variant}`), className)}
+      {...rest}
+    >
+      {buttons.map((button) => {
+        return (
+          <div
+            key="segmentedButtonGroupContainer"
```
Comment (lukeac123) [inline packages/lab/src/segmented-button-group/SegmentedButtonGroup.tsx:38]: not sure about what this key should be, is this fine ?

File: packages/lab/src/segmented-button-group/SegmentedButtonGroup.css [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+.saltSegmentedButtonGroup > *:not(:last-child)::after {
+  content: "";
+  width: var(--salt-size-border);
+  position: absolute;
+  padding-top: var(--salt-spacing-100);
+  padding-bottom: var(--salt-spacing-100);
+  background: var(--salt-palette-neutral-primary-separator);
+  right: calc(var(--salt-size-border) * -0.5);
```
Comment (lukeac123) [inline packages/lab/src/segmented-button-group/SegmentedButtonGroup.css:13]: Shift the separator so it sits directly between the two buttons

File: .changeset/sharp-flowers-relax.md [hunk 11]
```diff
@@ -0,0 +1,15 @@
+---
+"@salt-ds/lab": minor
+---
+
+Add `SegmentedButton` to labs
```
Comment (joshwooding) [inline .changeset/sharp-flowers-relax.md:5]: This needs updating.

File: packages/lab/src/__tests__/__e2e__/segmented-button-group/SegmentedButtonGroup.cy.tsx [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import { composeStories } from "@storybook/react";
+import * as dialogStories from "@stories/segmented-button-group/segmented-button-group.stories";
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/segmented-button-group/SegmentedButtonGroup.cy.tsx:2]: This name is wrong but I also don't see the value in having this test

File: packages/lab/stories/segmented-button-group/segmented-button-group.qa.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+  chromatic: { disableSnapshot: false },
+};
+
+export const Icons: StoryFn<SegmentedButtonGroupProps> = () => {
+  return (
+    <StackLayout>
+      {variants.map((variant) => {
+        return (
+          <SegmentedButtonGroup key={variant}>
+            <Button variant={variant}>
```
Comment (joshwooding) [inline packages/lab/stories/segmented-button-group/segmented-button-group.qa.stories.tsx:49]: We should follow our own guidance and make these accessible

File: packages/lab/stories/segmented-button-group/segmented-button-group.stories.tsx [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+  );
+};
+
+export const Icons: StoryFn<SegmentedButtonGroupProps> = () => {
+  return (
+    <StackLayout>
+      {variants.map((variant) => {
+        return (
+          <SegmentedButtonGroup key={variant}>
+            <Button variant={variant}>
```
Comment (joshwooding) [inline packages/lab/stories/segmented-button-group/segmented-button-group.stories.tsx:37]: These also need to be updated to follow our icon button guidance

**Review by joshwooding — COMMENTED** [review]
The separators don't have a background behind then so the colour from the page background affects their colour e.g.

image

File: packages/lab/src/__tests__/__e2e__/segmented-button-group/SegmentedButtonGroup.cy.tsx (hunk 12 above)
Comment (lukeac123) [reply packages/lab/src/__tests__/__e2e__/segmented-button-group/SegmentedButtonGroup.cy.tsx]: Is it not good just to make sure there are no errors when the component is rendered ?

File: packages/lab/src/segmented-button-group/SegmentedButtonGroup.css [hunk 15]
```diff
@@ -0,0 +1,24 @@
+.saltSegmentedButtonGroup {
+  display: flex;
+  flex-direction: row;
+  background: var(--button-background);
+  gap: var(--salt-size-border);
+}
+
+.saltSegmentedButtonGroup > *:not(:last-child)::after {
+  content: "";
+  width: var(--salt-size-border);
```
Comment (Fercas123) [inline packages/lab/src/segmented-button-group/SegmentedButtonGroup.css:10]: probably more of a design question but we have some separator variables in use for stack layout, i wonder if they are the same, if we should use them here too (they might be on a deprecating path, idk)

Comment (lukeac123) [reply packages/lab/src/segmented-button-group/SegmentedButtonGroup.css:10]: from what i can see  --stackLayout-separator-weight is equated to --salt-size-border ? Or is there another variable you're talking about ?

**Review by Fercas123 — APPROVED** [review]

**Review by lilyvc — APPROVED** [review]
lgtm

File: packages/lab/stories/segmented-button-group/segmented-button-group.stories.tsx [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+};
+
+export const Icons: StoryFn<SegmentedButtonGroupProps> = () => {
+  return (
+    <StackLayout>
+      {variants.map((variant) => {
+        return (
+          <SegmentedButtonGroup key={variant}>
+            <Button variant={variant} aria-label="Message">
+              <Tooltip content="Message">
```
Comment (joshwooding) [inline packages/lab/stories/segmented-button-group/segmented-button-group.stories.tsx:38]: Tooltips should be around the Button

Comment (lukeac123) [reply packages/lab/stories/segmented-button-group/segmented-button-group.stories.tsx:38]: Will fix in a follow up pr with the site docs and examples

**Review by joshwooding — APPROVED** [review]

**Pull request merged**